seismic/xcorqc/xcorqc.py

```diff
@@ -206,21 +206,15 @@ def xcorr2(tr1, tr2, sta1_inv=None, sta2_inv=None,
                 tr1_d = np.array(tr1_d_all[wtr1s:wtr1e], dtype=np.float32)
                 tr2_d = np.array(tr2_d_all[wtr2s:wtr2e], dtype=np.float32)
 
-                # detrend
-                tr1_d = spline(tr1_d, 2, 1000)
-                tr2_d = spline(tr2_d, 2, 1000)
+                # STEP 1: detrend
+                tr1_d = signal.detrend(tr1_d)
+                tr2_d = signal.detrend(tr2_d)
 
-                # zero-mean
+                # STEP 2: demean
```

Comment:
I thought the detrending in STEP 1, which subtracts a fitted line y = a*x + b, would have removed the mean already, because b is the mean term.

Reply:
This is Babak's special recipe :-).